p2p/validation/network_data_fetcher.go

```diff
@@ -0,0 +1,16 @@
+package validation
+
+import (
+	"github.com/ssvlabs/ssv-spec/types"
+)
+
+// Performs data fetching from the Network
+type NetworkDataFetcher interface {
+	ValidDomain(domain []byte) bool
+	CorrectTopic(committee []types.OperatorID, topic string) bool
+	GetCommitteeInfo(msgID types.MessageID) *CommitteeInfo
+	ExistingValidator(validatorPK types.ValidatorPK) bool
+	ActiveValidator(validatorPK types.ValidatorPK) bool
+	ValidatorLiquidated(validatorPK types.ValidatorPK) bool
+	ExistingCommitteeID(committeeID types.CommitteeID) bool
```

Comment:
For those methods we query the contract right?
And for the others we don't, right? So maybe should be 2 interfaces?

I also think we should add comments for interfaces methods, especially ones we only implement as test methods